app/containers/WriteReviewPage/index.js

```diff
@@ -9,33 +9,69 @@ import PropTypes from 'prop-types';
 import { connect } from 'react-redux';
 import { createStructuredSelector } from 'reselect';
 import { compose } from 'redux';
-import Select from 'react-select';
-import Header from 'components/Header';
-import PageSection from 'components/PageSection';
-import BorderBottomDiv from 'components/shared/BorderBottomDiv';
-import Stars from 'components/Stars';
 import injectReducer from 'utils/injectReducer';
 import injectSaga from 'utils/injectSaga';
 
-import 'react-select/dist/react-select.css';
-
-import { locationList, selectedLocationChange } from './actions';
-import { selectLocationList, selectSelectedLocationId } from './selectors';
+import { locationList, guideList, selectedLocationChange } from './actions';
+import { selectLocationList, selectGuideList, selectSelectedLocationId } from './selectors';
 import reducer from './reducer';
 import saga from './saga';
 
+import Helmet from 'react-helmet'
+
+import Header from 'components/Header';
+import PageSection from 'components/PageSection';
+import BorderBottomDiv from 'components/shared/BorderBottomDiv';
+import Select from 'components/Select';
+import Stars from 'components/Stars';
+import GuideReviewForm from 'components/GuideReviewForm';
+
 export class WriteReviewPage extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
 
   constructor(props) {
     super(props);
-
+    this.handleDetailsChange = this.handleDetailsChange.bind(this);
+    this.handleConfirmGuided = this.handleConfirmGuided.bind(this);
+    this.handleSelectGuideId = this.handleSelectGuideId.bind(this);
     this.handleSubmit = this.handleSubmit.bind(this);
+
+    // State should contain form data for easy validation and submit.
+    this.state = {
+      // Should figure out a way to pass selectedId props based on a review page we're on
+      selectedLocationId: props.selectedLocationId ? props.selectedLocationId : null,
+      tripRating: 0,
+      details: '',
+      selectedGuideId: props.selectedGuideId ? props.selectedGuideId : null,
+      guideRating: 0,
+    }
   }
 
   componentDidMount() {
+    // API call
     this.props.requestLocationList();
   }
 
+  handleDetailsChange(e) {
+    this.setState({ details: e.target.value });
```

Comment:
This will overwrite all state!! This is a big bug.

Repro: Click a star rating. Then fill out any text in the details section. It kills the star rating!

Please never call setState without copying current state. This is a good example for why I lean redux for things like this. It makes state harder to fuck up

Reply:
I've only seen this kind of thing update the specified state property and hadn't seen warnings about it till your comments. I will educate myself more on this.

Reply:
I believe `setState` is a merge, not an overwrite: https://github.com/facebook/react/blob/4a20ff26ecfe9bc66941d79f7fce2c67be8ee236/packages/react-dom/src/server/ReactPartialRenderer.js#L478

Reply:
also see https://stackoverflow.com/questions/23293626/setstate-vs-replacestate-in-react-js

Reply:
Makes no sense, then why is it overwriting the other sate

Reply:
I see though, good call

Reply:
This happens because the stars ratings under location isn't saved in any way, so making a change that updates the component will reset it to 0 because that's the value being passed in the code. I started working on a way to keep the ratings in state right after I submitted this PR, so it will be there very soon after this gets taken care of.